### PR TITLE
fix(menubar: init: get_current_page): list_spacing depends on presence of icon; also take cursor block width into consideration

### PR DESCRIPTION
**Pull request by actionless**

### CODE REVIEW

**Comment by Elv13** [comment]
Do you have a screenshot of before/after? Otherwise, it looks good to me

**Comment by actionless** [comment]
@Elv13  just see here: https://github.com/awesomeWM/awesome/blob/master/lib/awful/widget/common.lua#L76-L94
there are 3 x 4px spacings

and previous version of menbar had it hardcoded to be 3

however recently i've discovered (https://github.com/awesomeWM/awesome/blob/master/lib/awful/widget/common.lua#L184) what icon's margin got removed when icon is not present

**Review by Elv13 — DISMISSED** [review]

**Comment by actionless** [comment]
@Elv13 i was just looking on a final diff, every time i look once again on this, it's every time something new to fix

**Comment by actionless** [comment]
also `codecov` runs on lua 5.3, which skips tests/test-menubar, so it says what there is no coverage

**Comment by actionless** [comment]
BUMP

File: lib/menubar/init.lua [hunk 1]
```diff
@@ -160,7 +159,8 @@ local function label(o)
     return colortext(gstring.xml_escape(o.name), fg_color),
            bg_color,
            nil,
-           o.icon
+           o.icon,
+           o.icon and {icon_size=instance.geometry.height}
```
Comment (sclu1034) [inline lib/menubar/init.lua:163]: Can you add that to the function's docs as well?

**Review by sclu1034 — CHANGES_REQUESTED** [review]

File: lib/menubar/init.lua (hunk 1 above)
Comment (actionless) [reply lib/menubar/init.lua:163]: added

**Review by sclu1034 — APPROVED** [review]

File: lib/menubar/init.lua [hunk 2]
```diff
@@ -119,10 +119,9 @@ menubar.right_label = "▶▶"
 -- @tfield[opt="◀◀"] string left_label
 menubar.left_label = "◀◀"
 
--- awful.widget.common.list_update adds three times a margin of dpi(4)
--- for each item:
--- @tfield number list_interspace
-local list_interspace = theme.xresources.apply_dpi(4) * 3
+-- awful.widget.common.list_update adds spacing of dpi(4) between items.
+-- @tfield number list_spacing
+local list_spacing = theme.xresources.apply_dpi(4)
```
Comment (Aire-One) [inline lib/menubar/init.lua:124]: Do you think it would make sense to expose this field?

Comment (actionless) [reply lib/menubar/init.lua:124]: oh, wait, i double-checked, actually that spacing is hardcoded in awful.widgetcommon.list_update(), so here this variable only reflects that hardcoded value of common.list_update

Comment (Aire-One) [reply lib/menubar/init.lua:124]: For future references:

https://github.com/awesomeWM/awesome/blob/master/lib/awful/widget/common.lua#L82
https://github.com/awesomeWM/awesome/blob/master/lib/awful/widget/common.lua#L88
https://github.com/awesomeWM/awesome/blob/master/lib/awful/widget/common.lua#L89

**Review by Aire-One — APPROVED** [review]

**Review by Elv13 — DISMISSED** [review]

**Comment by actionless** [comment]
@Elv13 https://github.com/awesomeWM/awesome/pull/3160#issuecomment-827528799

**Review by Aire-One — DISMISSED** [review]

**Review by sclu1034 — APPROVED** [review]

**Comment by actionless** [comment]
finally made ci happy :rocket:

**Review by Aire-One — APPROVED** [review]

**Review by sclu1034 — APPROVED** [review]

**Comment by Elv13** [comment]
thanks!

**Pull request merged**